.github/workflows/build.yml

```diff
@@ -19,6 +19,53 @@ jobs:
       - name: unittest
         run: .build/run-unittest
 
+  # Builds and runs tests on Ubuntu 7.10.
+  # It ships with:
+  #
+  # gcc (GCC) 4.1.3 20070929 (prerelease) (Ubuntu 4.1.2-16ubuntu2)
+  # Copyright (C) 2006 Free Software Foundation, Inc.
+  #
+  # cmake version 2.4-patch 7
+  #
+  # This one is challenging:
+  #
+  # * It's impossible to run git clone inside this container (as old git will use ancient protocols not supported anymore
+  #   by GitHub, and also standard `actions/checkout@v3` will depend on NodeJS/npm, which are not available in this container).
+  #   To work around this, we start/stop container manually and use targeted execution of specific build/run commands only
+  #   inside the container, keeping git and networking stuff out.
+  #
+  # * For gcc, vast majority of C++11 features are missing from this release (many were added around GCC 4.3 and 4.4).
+  #   Some C99 features are also missing. C++98 support seems to be full-featued.
+  #
+  # * Modern GTest doesn't work in this environment, as GTest v1.12.x requires C++11 and modern GTest v1.13+ requires C++14.
+  #   To work around this, we supply nano substitute of GTest (tests/gtest-nano.h), partially compatible with API.
+  #
+  # * CMake is way too old to be usable without major rewrite of all the CMakeLists.txt to use ancient/deprecated syntax.
+  #   To work around this, we use plain GNU make(1) and supply custom Makefile.gcc4.
+  linux_ubuntu_7_10:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: docker pull
+        run: |
+          docker pull icomputer7/ancient-ubuntu-docker:gutsy
+      - name: docker create start
+        run: |
+          docker create --name u710 --label u710 --workdir /w -v "${{ github.workspace }}":"/w" --entrypoint "tail" icomputer7/ancient-ubuntu-docker:gutsy "-f" "/dev/null"
+          docker start u710
```

Comment:
I don't like the way `${{ github.workspace }}` is injected there. You're letting the GitHub runner to interpolate the default working directory to some text that will be later used as a shell command, so the runner is clueless about shell syntax and what are the characters with special meaning for the shell in the double quoted string literal (`$`, `` ` ``, `\`, `"` and possibly others) that must not appear in the `github.workspace`. Maybe in this particular case we don't expect that the directory GitHub came up will be this crazy, but to me it's wrong anyway (and I generally also find it easier and much less error-prone to only ever use the safe and reliable patterns than having to analyze every time whether the unsafe pattern is "safe enough" for a given use case or not).

The simplest solution I see is to let the shell do the variable expansion and use the `GITHUB_WORKSPACE` environment variable instead (see [Default environment variables](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables)):

```suggestion
      - name: docker create start
        run: |
          docker create --name u710 --label u710 --workdir /w -v "$GITHUB_WORKSPACE":"/w" --entrypoint "tail" icomputer7/ancient-ubuntu-docker:gutsy "-f" "/dev/null"
          docker start u710
```

Reply:
Makes sense, let's give it a try. In all fairness, though, I believe you're right and GitHub will never use anything which needs complicated quoting in this place.